### PR TITLE
Roll Dart SDK to 2.0.0-dev.16.0, along with associated package updates

### PR DESCRIPTION
**Pull request by leafpetersen**
[Edited: rolling to dev.16.0 instead of dev.15.0 or dev.14.0 to work around asset validation issues and pub issues]

This is the final step in updating the Dart SDK in the flutter framework.  No code changes except:
  - remove the --assert-initializer flags.  
  - explicitly pass allowTrailingOptions: false to new ArgParser instances to account for breaking change in the ArgParser API

Main changes are:
  - Update the sdk version to 2.0.0-dev.16.0
  - Upgrade the pinned versions of packages to pull in the latest versions

### CODE REVIEW

**Comment by leafpetersen** [comment]
Okay, this is the last step. 

@Hixie @mit-mit Do I (or we) need to do some breaking change messaging before we land this, since this will add the additional "dynamic as bottom" warnings to flutter analyze?  If so, can you remind me what your process is?

**Comment by leafpetersen** [comment]
Looks like some real test failures, I think from the args package roll.  Looking into them now.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -61,6 +62,10 @@ class FlutterOptions {
 }
 
 abstract class FlutterCommand extends Command<Null> {
+  @override
+  ArgParser get argParser => _argParser;
+  final ArgParser _argParser = new ArgParser(allowTrailingOptions: false);
```
Comment (leafpetersen) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: The default value of allowTrailingOptions has flipped.  This restores the old behavior.

**Comment by leafpetersen** [comment]
Ok, I fixed an issue introduced by the args roll.  I'm seeing failures that look like this on travis though:

```
⏩ RUNNING: cd .; bin/flutter analyze --flutter-repo --watch --benchmark
Analyzing Flutter repository...
warning • The asset packages/flutter_gallery_assets/ali_connors.png does not exist. • dev/benchmarks/complex_layout/pubspec.yaml:69:7
warning • The asset packages/flutter_gallery_assets/ali_connors_sml.png does not exist. • dev/benchmarks/complex_layout/pubspec.yaml:70:7
warning • The asset packages/flutter_gallery_assets/top_10_australian_beaches.png does not exist. • dev/benchmarks/complex_layout/pubspec.yaml:71:7
warning • The asset packages/flutter_gallery_assets/animated_flutter_lgtm.gif does not exist. • examples/flutter_gallery/pubspec.yaml:73:7
```

Can someone give me some context about those error messages?  I see a reference to those assets in `pubspec.yaml`, which doesn't seem to have changed, I don't see those files anywhere in the repo.  How are they supposed to come into being?

**Comment by devoncarew** [comment]
@leafpetersen, I believe those warnings were introduced by a recent change by @bwilkerson to add validation to the flutter section of the pubspec.yaml file. I have not looked into the specific issue, so don't know if it's found an actual issue or there's a problem with the analysis.

**Comment by devoncarew** [comment]
`flutter analyze` may not be written to apply the correct cwd from the POV of the lint.

**Comment by devoncarew** [comment]
You could choose to bump to something slightly before `2.0.0-dev.15.0` - this analysis change was fairly recent.

**Comment by mit-mit** [comment]
@devoncarew  @bwilkerson great to hear we have asset validation! But we should probably send out a breaking change notification for it, see for example: https://groups.google.com/forum/#!topic/flutter-dev/FJHSnAnhx1I

**Comment by mit-mit** [comment]
@leafpetersen thanks! Yes, we probably need a breaking change notification for fuzzy arrows and related typing changes. See the example I linked to above.

**Comment by Hixie** [comment]
https://flutter.io/design-principles/#handling-breaking-changes is the policy for breaking changes, yes please do follow it

**Comment by devoncarew** [comment]
Brian has committed a fix for the pubspec lint (of `packages/` assets), so that'll be in the next dev sdk.

**Comment by leafpetersen** [comment]
Ok, we may have to roll forward rather than backwards.  I tried backing off to dev.14.0 but ran into this:  https://github.com/dart-lang/pub/issues/1771 .  The failure mode is a bug in pub, but my suspicion is that there is a real version issue there which will require us to roll forward to dev.15.0 or later.

**Comment by Hixie** [comment]
Rolling forward is probably simplest, since the breaking change policy suggests a one week waiting period before landing the change anyway (to gather any feedback).

**Comment by leafpetersen** [comment]
Breaking change email went out here:  https://groups.google.com/forum/#!topic/flutter-dev/E22RdKAYjs8 .

**Comment by leafpetersen** [comment]
Ok, we have a green build.  @Hixie when you get a chance can you take a look?  Assuming you're good with this, I'll plan on waiting a couple more days in case we get any more feedback from the breaking change announcement, then merge.

**Comment by Hixie** [comment]
LGTM but before you land this please wait for https://github.com/flutter/flutter/pull/13954 to land and then rerun the script to update the dependencies. That PR fixes a bug in the script.

**Comment by mit-mit** [comment]
I suggest we land this tomorrow, Wednesday. That gives us a few days to handle any breaking change issues in "the wild" before the weekend.

**Comment by leafpetersen** [comment]
SGTM.  I'll plan on doing a final merge of the branch tomorrow morning, and assuming all is still green merge to master.

File: packages/flutter/lib/src/material/ink_ripple.dart [hunk 2]
```diff
@@ -20,8 +20,6 @@ const Duration _kCancelDuration = const Duration(milliseconds: 75);
 // The fade out begins 300ms after the _fadeOutController starts. See confirm().
 const double _kFadeOutIntervalStart = 300.0 / 450.0;
 
-const double _kRippleConfirmedVelocity = 1.0; // logical pixels per millisecond
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/material/ink_ripple.dart]: cc @HansMuller who added this recently, but if it's unused then removing it should be fine...

**Comment by Hixie** [comment]
#13954 has landed now, please remember to rerun the `flutter update-packages --force-upgrade` script before landing.

**Comment by leafpetersen** [comment]
Yup, ran it, and just ran it again, still clean.

**Pull request merged**